connector_magento/models/magento_backend/common.py

```diff
@@ -0,0 +1,418 @@
+        'store view.'
+    )
+    is_multi_company = fields.Boolean(
+        string='Is Backend Multi-Company',
+        help="If this flag is set, it is possible to choose warehouse at each "
+        "level. "
+        "When import partner, ignore company_id if this flag is set.",
+    )
+
+    _sql_constraints = [
+        ('sale_prefix_uniq', 'unique(sale_prefix)',
+         "A backend with the same sale prefix already exists")
+    ]
+
+    @api.multi
+    def check_magento_structure(self):
+        """ Used in each data import.
+
+        Verify if a website exists for each backend before starting the import.
+        """
+        for backend in self:
+            websites = backend.website_ids
+            if not websites:
+                backend.synchronize_metadata()
+        return True
+
+    @contextmanager
+    @api.multi
+    def work_on(self, model_name, **kwargs):
+        self.ensure_one()
+        lang = self.default_lang_id
+        if lang.code != self.env.context.get('lang'):
+            self = self.with_context(lang=lang.code)
+        magento_location = MagentoLocation(
+            self.location,
+            self.username,
+            self.password,
+            use_custom_api_path=self.use_custom_api_path
+        )
+        if self.use_auth_basic:
+            magento_location.use_auth_basic = True
+            magento_location.auth_basic_username = self.auth_basic_username
+            magento_location.auth_basic_password = self.auth_basic_password
+        # We create a Magento Client API here, so we can create the
+        # client once (lazily on the first use) and propagate it
+        # through all the sync session, instead of recreating a client
+        # in each backend adapter usage.
+        with MagentoAPI(magento_location) as magento_api:
+            _super = super(MagentoBackend, self)
+            # from the components we'll be able to do: self.work.magento_api
+            with _super.work_on(
+                    model_name, magento_api=magento_api, **kwargs) as work:
+                yield work
+
+    @api.multi
+    def add_checkpoint(self, record):
+        self.ensure_one()
+        record.ensure_one()
+        return add_checkpoint(self.env, record._name, record.id,
+                              self._name, self.id)
+
+    @api.multi
+    def synchronize_metadata(self):
+        try:
+            for backend in self:
+                for model_name in ('magento.website',
+                                   'magento.store',
+                                   'magento.storeview'):
+                    # import directly, do not delay because this
+                    # is a fast operation, a direct return is fine
+                    # and it is simpler to import them sequentially
+                    self.env[model_name].import_batch(backend)
+            return True
+        except Exception as e:
+            _logger.error(e.message, exc_info=True)
```

Comment:
This call will fail because the Exception `e` doesn't have an attribute called `message`

Reply:
Changes Done. Now Uses the ustr method from Odoo base to print the exception as text.